src/pip/_internal/wheel.py

```diff
@@ -620,6 +620,17 @@ def supported(self, tags=None):
         return bool(set(tags).intersection(self.file_tags))
 
 
+def _is_like_egg_info(
+        s, _egg_info_re=re.compile(r'([a-z0-9_.]+)-([a-z0-9_.!+-]+)', re.I)):
+    """Determine whether the string looks like an egg_info.
+
+    :param s: The string to parse. E.g. foo-2.1
+    """
+    if not _egg_info_re.search(s):
```

Comment:
Also, I would reverse the True and False cases for simplicity, or even cast to boolean.